Sources/SWBCore/Specs/Tools/LinkerTools.swift

```diff
@@ -1549,14 +1555,20 @@ public func discoveredLinkerToolsInfo(_ producer: any CommandProducer, _ delegat
                 if let match = try gnuLD.compactMap({ try $0.firstMatch(in: String(decoding: executionResult.stdout, as: UTF8.self)) }).first {
                     return DiscoveredLdLinkerToolSpecInfo(linker: .gnuld, toolPath: toolPath, toolVersion: try Version(String(match.output.version)), architectures: Set())
                 }
-
                 let goLD = [
                     #/GNU gold version (?<version>[\d.]+)-.*/#,
                     #/GNU gold \(GNU Binutils.*\) (?<version>[\d.]+)/#,
                 ]
                 if let match = try goLD.compactMap({ try $0.firstMatch(in: String(decoding: executionResult.stdout, as: UTF8.self)) }).first {
                     return DiscoveredLdLinkerToolSpecInfo(linker: .gold, toolPath: toolPath, toolVersion: try Version(String(match.output.version)), architectures: Set())
                 }
+                // link.exe has no option to simply dump the version, but if you pass an invalid argument, the program will dump a header that contains the version.
```

Comment:
This is not great, but I can't think of anything better

Reply:
FWIW, invoking with no arguments will also print the version header and is a valid invocation (prints help info), so it is maybe "cleaner" than intentionally passing an invalid argument.